src/toil/__init__.py

```diff
@@ -184,6 +184,30 @@ def checkDockerImageExists(appliance):
         return requestCheckDockerIo(registryName=registryName, tag=tag)
 
 
+def makeApplianceImageNotFound(name, tag, url, statusCode):
```

Comment:
This seems to be a custom error, I'd like to argue that it should inherit `ImageNotFound` following the conventions of other custom errors in Toil.  Something roughly like the following:

```
class ApplianceNotFoundException(ImageNotFound):
    def __init__(self, name, tag, url, statusCode):
        msg = ("The docker image that TOIL_APPLIANCE_SELF specifies (%s) produced "
               "a nonfunctional manifest URL (%s). The HTTP status returned was %s. "
               "The specifier is most likely unsupported or malformed.  "
               "Please supply a docker image with the format: "
               "'<websitehost>.io/<repo_path>:<tag>' or '<repo_path>:<tag>' "
               "(for official docker.io images).  Examples: "
               "'quay.io/ucsc_cgl/toil:latest', 'ubuntu:latest', or "
               "'broadinstitute/genomes-in-the-cloud:2.0.0'."
               "" % (name + ':' + tag, url, str(statusCode)))
        super(ApplianceNotFoundException, self).__init__(msg)
```

Reply:
This isn't a custom error type; it's a function that returns ImageNotFound errors. I just needed one place to put the template string logic that was common to both image checking functions.

Reply:
Unless you think we really do want a custom error type for exactly this particular case, which always has a message that looks like that? It seems a bit overkill for something that would be thrown in pretty much one situation ever.

Reply:
A class that is raised to return an error is equivalent in function to a custom error.  Two is not "pretty much one", and I don't see having a more explicitly named error as necessarily a bad thing.

Reply:
Something that is raised to return an error seems equivalent in function to a custom error.

Two may be "pretty much one" sometimes, but isn't the fact that it's used in more than one situation the impetus behind writing the function in the first place?

Also, having a more explicitly named error in general isn't necessarily a bad thing.

Reply:
The type of the error being raised in this case is still ImageNotFound.

The syntax at the point of use is the same between a function call and a new type. If you really want the class version I can do that instead.

Reply:
It's true that it's an ImageNotFound error.  It's also a RuntimeError.  These are varying degrees of specificity.  I think it's fine to keep it as an ImageNotFound error, rather than making it more specific.  A more specific error name was just a suggestion.

The error class, however, is a convention in Toil (and in python in general).  See PEP8 guidelines for creating exceptions:

https://www.python.org/dev/peps/pep-0008/#exception-names

I'm trying to argue for consistency in toil.  I can link all of the exception classes that are consistently written throughout Toil if it will convince you that it's a good idea to follow convention, most of which simply hold a custom string error message just as your function is doing.